compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1447,6 +1454,13 @@ impl<'a> Parser<'a> {
         self.maybe_recover_from_bad_qpath(expr, true)
     }
 
+    fn parse_invisibles_expr(&mut self) -> PResult<'a, P<Expr>> {
+        self.expect(&token::OpenDelim(Delimiter::Invisible(InvisibleSource::ExprMv)))?;
+        let expr = self.parse_expr()?;
```

Comment:
I tried changing this function like so:
```rust
     pub fn parse_invisibles_expr(&mut self) -> PResult<'a, P<Expr>> {
+        let lo = self.token.span;
         self.expect(&token::OpenDelim(Delimiter::Invisible(InvisibleSource::ExprMv)))?;
-        let expr = self.parse_expr()?;
+        let mut expr = self.parse_expr()?;
         self.expect(&token::CloseDelim(Delimiter::Invisible(InvisibleSource::ExprMv)))?;
+        expr.span = lo.to(self.prev_token.span);
         self.maybe_recover_from_bad_qpath(expr, true)
     }
```
I get about 50 new ui test failures. Interestingly, it seems for quite a few of them it's because the error messages have gotten *better*. E.g. some of the test I previously blessed have had their output revert to the original output. And plenty that weren't affected before now are. I haven't had time to go through them all carefully to understand what's going on.

Strangely enough, this change even caused one test to stop compiling! It is `src/test/ui/fmt/format-args-capture.rs`. The new error is this:
```
error: there is no argument named `foo`
  --> src/test/ui/fmt/format-args-capture.rs:58:9
   |
58 |         panic!("{}-{bar}-{foo}", "positional", bar = "named");
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: did you intend to capture a variable `foo` from the surrounding scope?
   = note: to avoid ambiguity, `format_args!` cannot capture variables when the format string is expanded from a macro
   = note: this error originates in the macro `$crate::panic::panic_2015` (in Nightly builds, run with -Z macro-backtrace for more info)
```
which I don't really understand.

Reply:
>Strangely enough, this change even caused one test to stop compiling!

That may happen if the `SyntaxContext` part of the span changes in an unexpected way, I'll check later.

Reply:
`fn parse_invisibles_expr` is not the reasons, it does everything right, the `lo` span is not set correctly in some other places.
See e.g. https://github.com/petrochenkov/rust/commit/68c3423a7857585f06b5ee5958890343197288e1 that fixes some of the spans.

Reply:
Thanks for the fix! I have incorporated it.